### PR TITLE
[SYCL][FPGA] Improve intel-fpga-local tests

### PR DESCRIPTION
**Pull request by vmaksimo**
This PR introduces changes to both sema and codegen tests.

Changes for codegen test:
* Renamed all FileCheck labels to avoid alphabetical changes in future
* Change all variables and function names to meaningful ones
* Remove duplicate test cases

Changes for sema test:
* Change all variables and function names to meaningful ones

### CODE REVIEW

File: clang/test/SemaSYCL/intel-fpga-local.cpp [hunk 1]
```diff
@@ -91,185 +91,187 @@ void foo1()
...
+  //CHECK: VarDecl{{.*}}doublepump_mlab
   //CHECK: IntelFPGADoublePumpAttr
   //CHECK: IntelFPGAMemoryAttr{{.*}}MLAB{{$}}
   [[intelfpga::doublepump]]
   [[intelfpga::memory("MLAB")]]
-  unsigned int v_fourteen[64];
+  unsigned int doublepump_mlab[64];
 
-  //CHECK: VarDecl{{.*}}v_fifteen
+  //CHECK: VarDecl{{.*}}mlab_doublepump
```
Comment (AGindinson) [inline clang/test/SemaSYCL/intel-fpga-local.cpp]: I don't recall if there's a reason to do both `mlab_doublepump` and `doublepump_mlab`. Could you please bring it up/add it as a comment, or remove the extra case?

File: clang/test/SemaSYCL/intel-fpga-local.cpp [hunk 2]
```diff
@@ -431,152 +433,151 @@ void foo1()
...
 
   //expected-error@+1{{bank_bits must be consecutive}}
   [[intelfpga::bank_bits(3,3,4), intelfpga::bankwidth(4)]]
-  unsigned int bb_five[4];
+  unsigned int bb_noncons[4];
 
   //expected-error@+1{{bank_bits must be consecutive}}
   [[intelfpga::bank_bits(1,3,4), intelfpga::bankwidth(4)]]
-  unsigned int bb_six[4];
+  unsigned int bb_noncons_2[4];
```
Comment (AGindinson) [inline clang/test/SemaSYCL/intel-fpga-local.cpp]: For the sake of consistency:
```suggestion
  unsigned int bb_noncons_1[4];
```

File: clang/test/CodeGenSYCL/intel-fpga-local.cpp [hunk 3]
```diff
@@ -1,210 +1,179 @@
...
-  // CHECK-DEVICE: %[[FIELD9:.*]] = getelementptr inbounds %struct.{{.*}}.foo_two{{.*}}
-  // CHECK-DEVICE: call i32* @llvm.ptr.annotation.p0i32{{.*}}%[[FIELD9]]{{.*}}[[ANN13]]
-  s1.f9 = 0;
-  // CHECK-DEVICE: %[[FIELD10:.*]] = getelementptr inbounds %struct.{{.*}}.foo_two{{.*}}
-  // CHECK-DEVICE: call i32* @llvm.ptr.annotation.p0i32{{.*}}%[[FIELD10]]{{.*}}[[ANN14]]
-  s1.f10 = 0;
-  // CHECK-DEVICE: %[[FIELD11:.*]] = getelementptr inbounds %struct.{{.*}}.foo_two{{.*}}
-  // CHECK-DEVICE: call i32* @llvm.ptr.annotation.p0i32{{.*}}%[[FIELD11]]{{.*}}[[ANN15]]
-  s1.f11 = 0;
+void attr_on_static(int a) {
```
[the remark this answers is not in the document]
Comment (vmaksimo) [reply clang/test/CodeGenSYCL/intel-fpga-local.cpp]: Good point, will fix this

File: clang/test/SemaSYCL/intel-fpga-local.cpp (hunk 1 above)
Comment (vmaksimo) [reply clang/test/SemaSYCL/intel-fpga-local.cpp]: Will remove duplicating case

File: clang/test/SemaSYCL/intel-fpga-local.cpp (hunk 2 above)
Comment (vmaksimo) [reply clang/test/SemaSYCL/intel-fpga-local.cpp]: let's count variable names from 1, not from 0? 😏

File: clang/test/CodeGenSYCL/intel-fpga-local.cpp [hunk 4]
```diff
@@ -232,12 +202,9 @@ __attribute__((sycl_kernel)) void kernel_single_task(Func kernelFunc) {
 
 int main() {
   kernel_single_task<class kernel_function>([]() {
-    foo();
-    bar();
-    baz();
-    qux(42);
-    size_info();
-    bankbits();
+    attr_on_static();
```
Comment (mlychkov) [inline clang/test/CodeGenSYCL/intel-fpga-local.cpp]: Maybe to rename it to attrs_on_static() to be inline with other functions name?

**Pull request merged**